pkg/webhooks/subscription/subscription.go

```diff
@@ -0,0 +1,182 @@
+package subscription
+
+import (
+	"encoding/json"
+	"net/http"
+	"sync"
+
+	responsehelper "github.com/openshift/managed-cluster-validating-webhooks/pkg/helpers"
+	"github.com/openshift/managed-cluster-validating-webhooks/pkg/webhooks/utils"
+	"k8s.io/api/admission/v1beta1"
+	admissionregv1 "k8s.io/api/admissionregistration/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	logf "sigs.k8s.io/controller-runtime/pkg/runtime/log"
+	admissionctl "sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+)
+
+const (
+	WebhookName    string = "subscription-validation"
+	LoggingSubName string = "cluster-logging"
+	ESSubName      string = "elasticsearch-operator"
+	Channel        string = "4.5"
+)
+
+var (
+	privilegedUsers = []string{"kube:admin", "system:admin", "system:serviceaccount:openshift-authentication:oauth-openshift"}
```

Comment:
> `system:serviceaccount:openshift-authentication:oauth-openshift`

Does the OAUTH operator interact with Subscription objects?

Reply:
I'm not sure. I ripped the privilegedUsers var out of some other webhook to save time. I'm ok to rip that bit out. I think there may be some service account we want to account for, but I need to do some manual testing to figure that out

Reply:
That's actually a good question. I'm not sure what's responsible for updating this in-cluster. My first guess would be OLM.

Reply:
FTR, turns out the SA we needed to add was system:serviceaccount:kube-system:generic-garbage-collector. This allows the logging stack to be deleted